machine/v1beta1/types_vsphereprovider.go

```diff
@@ -87,12 +87,124 @@ type NetworkSpec struct {
 	Devices []NetworkDeviceSpec `json:"devices"`
 }
 
+type DHCPState string
+
+const (
+	EnabledState  DHCPState = "Enabled"
+	DisabledState DHCPState = "Disabled"
+)
+
+// AddressesFromPool is an IPAddressPool that will be used to create
+// IPAddressClaims for fulfillment by an external controller.
+type AddressesFromPool struct {
+	Group string `json:"group"`
+
+	// Kind is the type of resource being referenced
+	Resource string `json:"resource"`
+
+	// Name is the name of resource being referenced
+	Name string `json:"name"`
+}
+
 // NetworkDeviceSpec defines the network configuration for a virtual machine's
 // network device.
 type NetworkDeviceSpec struct {
 	// NetworkName is the name of the vSphere network to which the device
 	// will be connected.
-	NetworkName string `json:"networkName"`
+	// +optional
+	NetworkName string `json:"networkName,omitempty"`
+
+	// DeviceName may be used to explicitly assign a name to the network device
+	// as it exists in the guest operating system.
+	// +optional
+	DeviceName string `json:"deviceName,omitempty"`
+
+	// DHCP4 is a flag that indicates whether or not to use DHCP for IPv4
+	// on this device.
+	// If enabled, then IPAddrs should not contain any IPv4 addresses and
+	// AddressesFromPools must be empty.
```

Comment:
Need to explain valid values, `// Valid values are Enabled and Disabled. The default value is Enabled.`